ipld/kamt/src/ext.rs

```diff
@@ -16,16 +16,16 @@ use crate::{Error, HashedKey};
 /// to the next non-empty `Node`.
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Default)]
 pub(crate) struct Extension {
-    /// Number of bits consumed from the path between the `Node` containing the `Link`
-    /// and the node the `Link` is pointing to the next `Node`. It might be less than
-    /// the length of the slice of the path in the extension, for example if the bit
-    /// width is 3 and we consumed only 6 bits out of 8, which is the length of a byte.
+    /// The length (in bits) of the extension between the `Node` containing the `Link`
+    /// and the node the `Link` is pointing to. It might be less than the length of the
+    /// slice of the path in the extension, for example if the bit width is 3 and we
+    /// consumed only 6 bits out of 8, which is the length of a byte.
     ///
     /// The `bit_width` of a KAMT can at most be 8 (`HashBits::next` doesn't work with more),
     /// resulting in a maximum of 256 slots. The minimum `bit_width` is 1, which would be
     /// a binary tree. Because the root will take at least 1 bit from the hashed key,
     /// we know the extension can consume at most 255 bits, which should fit in a `u8`.
-    consumed: u8,
+    length: u32,
```

Comment:
Due to alignment, this almost certainly won't use more memory (although I guess we're removing a potential niche for discriminants?).

Reply:
Using `u32` instead of `u8`? I don't fully understand the alignment; are you saying a single byte is stored as 4 bytes? 

The main reason I used `u8` is because the `Hamt` and the `Kamt` both only support `bit_width` up to 8, which can result in at maximum 256 slots, which is exactly what `u8` can represent. Using a larger data type for indexes and the number of bits consumed is actively misleading IMO.

Reply:
I didn't want to distrub the `HashBits` to take and `u8` in `next`, but it would be more befitting the domain.

Reply:
If you really want to use `u32` here then I think the docstring should explain why, because currently it argues that the data fits into a `u8`, so anyone coming at a later time would be scratching their heads why it's not a `u8` then, or maybe change it back because they don't have intimate knowledge of how many instructions it takes in Wasm to add these numbers up.

Reply:
Removed.

But also, think of it this way: you had to write a long comment on why it was ok to use a u8 here. But using a u32 means we don't even need to talk about it.